src/Workspaces/Core/Portable/Workspace/Solution/ProjectInfo.cs

```diff
@@ -211,17 +213,17 @@ public sealed class ProjectInfo
             string name,
             string assemblyName,
             string language,
-            string filePath,
-            string outputFilePath,
-            CompilationOptions compilationOptions,
-            ParseOptions parseOptions,
-            IEnumerable<DocumentInfo> documents,
-            IEnumerable<ProjectReference> projectReferences,
-            IEnumerable<MetadataReference> metadataReferences,
-            IEnumerable<AnalyzerReference> analyzerReferences,
-            IEnumerable<DocumentInfo> additionalDocuments,
+            string? filePath,
```

Comment:
I assume those annotations won't interfere with backcompat?

Reply:
I'm generally marking things as nullable or not nullable based on their actual behavior to avoid "back compat" issues in that sense, except in a few places where it's a clear bug. For example, there's some APIs that *should* take a null, they do the wrong thing today, but I'm marking it as nullable because that's what the contract should have been and filing a bug to track actually fixing it.

Reply:
No matter what though, these are all just compile-time stuff; they have no runtime behavior changes.